### PR TITLE
Rename `QueuingContext` to `QueuingManager`

### PR DESCRIPTION
**Pull request by albi3ro**
This PR renames `QueuingContext` to `QueuingManager` to better reflect its role in keeping track of the active queue and placing queueable objects into the current active queue. Especially as `QueuingContext` is no longer a context.

The class is also no longer imported top-level, as it should be purely developer-facing.

`__getattr__` methods were added to `pennylane/queuing.py` and `pennylane/__init__.py` to help developers accommodate this name change.

This PR is a follow on to #2794  and must be merged after it.

### CODE REVIEW

File: pennylane/measurements.py [hunk 1]
```diff
@@ -489,7 +489,7 @@ def expand(self):
 
         return tape
 
-    def queue(self, context=qml.QueuingContext):
+    def queue(self, context=qml.queuing.QueuingManager):
```
Comment (antalszava) [inline pennylane/measurements.py:492]: Can this work?
```suggestion
    def queue(self, context=qml.QueuingManager):
```

File: pennylane/queuing.py [hunk 2]
```diff
@@ -105,8 +112,7 @@ def update_info(cls, obj, **kwargs):
         Args:
             obj: the object with metadata to be updated
         """
-        if cls.recording():
-            cls.active_context().update_info(obj, **kwargs)
+        return cls.active_context().update_info(obj, **kwargs) if cls.recording() else None
```
Comment (antalszava) [inline pennylane/queuing.py:115]: How come this method (and some other methods) have been changed to return objects? This seems out of scope here.

File: pennylane/transforms/qfunc_transforms.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def wrapper(*args, **kwargs):
         active_tape = qml.tape.get_active_tape()
 
         if active_tape is not None:
-            with active_tape.stop_recording(), active_tape.__class__() as tape:
+            with active_tape.stop_recording(), active_tape.__class__() as tape:  # pylint: disable=not-context-manager
```
Comment (antalszava) [inline pennylane/transforms/qfunc_transforms.py:71]: Hmm, should the context manager block be changed here? :thinking: How come the linting warning started to show?

**Review by antalszava — COMMENTED** [review]
A couple of questions/comments, but looks overall good so far. :+1:

File: pennylane/queuing.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/queuing.py:115]: Maybe an out-of-scope change.  Just a minor cleanup that my IDE was telling me to do.  Exact same behaviour, just a little more compact.

File: pennylane/transforms/qfunc_transforms.py (hunk 3 above)
Comment (albi3ro) [reply pennylane/transforms/qfunc_transforms.py:71]: I have no idea why this linting warning appeared, hence why I just silenced it. Problem might be fixed in #3068  ?

File: tests/test_queuing.py [hunk 4]
```diff
@@ -134,20 +153,20 @@ def test_get_info_none(self):
...
         """Test that update_info correctly updates an annotation"""
         A = qml.RZ(0.5, wires=1)
 
         with AnnotatedQueue() as q:
             q.append(A, inv=True)
-            assert QueuingContext.get_info(A) == {"inv": True}
+            assert QueuingManager.get_info(A) == {"inv": True}
 
-            qml.QueuingContext.update_info(A, key="value1")
+            qml.queuing.QueuingManager.update_info(A, key="value1")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:166]:
```suggestion
            QueuingManager.update_info(A, key="value1")
```

File: tests/test_queuing.py [hunk 5]
```diff
@@ -134,20 +153,20 @@ def test_get_info_none(self):
...
 
         # should pass silently because no longer recording
-        qml.QueuingContext.update_info(A, key="value2")
+        qml.queuing.QueuingManager.update_info(A, key="value2")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:169]:
```suggestion
        QueuingManager.update_info(A, key="value2")
```

File: tests/test_queuing.py [hunk 6]
```diff
@@ -173,9 +192,9 @@ def test_safe_update_info_queued(self):
         with AnnotatedQueue() as q:
             q.append(op, key="value1")
             assert q.get_info(op) == {"key": "value1"}
-            qml.QueuingContext.safe_update_info(op, key="value2")
+            qml.queuing.QueuingManager.safe_update_info(op, key="value2")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:195]:
```suggestion
            QueuingManager.safe_update_info(op, key="value2")
```

File: tests/test_queuing.py [hunk 7]
```diff
@@ -173,9 +192,9 @@ def test_safe_update_info_queued(self):
...
 
-        qml.QueuingContext.safe_update_info(op, key="no changes here")
+        qml.queuing.QueuingManager.safe_update_info(op, key="no changes here")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:197]:
```suggestion
        QueuingManager.safe_update_info(op, key="no changes here")
```

File: tests/test_queuing.py [hunk 8]
```diff
@@ -187,8 +206,8 @@ def test_safe_update_info_not_queued(self):
         op = qml.RX(0.5, wires=1)
 
         with AnnotatedQueue() as q:
-            qml.QueuingContext.safe_update_info(op, key="value2")
-        qml.QueuingContext.safe_update_info(op, key="no changes here")
+            qml.queuing.QueuingManager.safe_update_info(op, key="value2")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:209]:
```suggestion
            QueuingManager.safe_update_info(op, key="value2")
```

File: tests/test_queuing.py [hunk 9]
```diff
@@ -187,8 +206,8 @@ def test_safe_update_info_not_queued(self):
...
+        qml.queuing.QueuingManager.safe_update_info(op, key="no changes here")
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:210]:
```suggestion
        QueuingManager.safe_update_info(op, key="no changes here")
```

File: tests/test_queuing.py [hunk 10]
```diff
@@ -12,33 +12,52 @@
...
 import pennylane as qml
 import numpy as np
 
-from pennylane.queuing import AnnotatedQueue, AnnotatedQueue, QueuingContext, QueuingError
+from pennylane.queuing import AnnotatedQueue, AnnotatedQueue, QueuingManager, QueuingError
 
 
-class TestQueuingContext:
-    """Test the logic associated with the QueuingContext class."""
+def test_top_level_import_warning():
```
Comment (AlbertMitjans) [inline tests/test_queuing.py:26]: For some reason this test fails.

File: pennylane/queuing.py (hunk 2 above)
Comment (antalszava) [reply pennylane/queuing.py:115]: Well, previously we always returned `None`. Now, this may return an object that is `not None` depending on what `cls.active_context().update_info(obj, **kwargs)` returns.

Overall, I think it's better to revert these changes. The new version doesn't seem to make too much sense in our logic. What reason did the IDE mention?

Comment (albi3ro) [reply pennylane/queuing.py:115]: I think it was mostly complaining about the `get_info` method. Then I just realized all the other methods have basically the same form and changed them too.

I've reverted all the methods other than `get_info`.

File: tests/test_queuing.py (hunk 10 above)
Comment (albi3ro) [reply tests/test_queuing.py:26]: I was going to stop importing the object top-level, but then I realized I wanted easy access to `qml.QueuingManager.active_context()` and `qml.QueuingManger.stop_recording()`.  So it is once again imported top-level.

File: pennylane/queuing.py (hunk 2 above)
Comment (antalszava) [reply pennylane/queuing.py]: Thanks. What issue did it raise with `get_info`?

**Review by antalszava — COMMENTED** [review]
Thanks for the responses! I'm still curious about what the IDE mentioned wrt. `get_info`. The change doesn't seem too drastic, but I fear it can lead to unexpected behaviour that we don't yet foresee.

File: pennylane/queuing.py (hunk 2 above)
Comment (AlbertMitjans) [reply pennylane/queuing.py]: This refactor can be applied to `get_info`, because this method has a `return` statement inside and outside the `if`.

**Review by AlbertMitjans — APPROVED** [review]
Good job!

File: pennylane/queuing.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/queuing.py]: Exactly.  `get_info` was returning values in both cases.

Comment (antalszava) [reply pennylane/queuing.py]: :100: Thanks!

**Review by antalszava — APPROVED** [review]
:clap: :clap: :clap:
